cura/Settings/MachineManager.py

```diff
@@ -1531,3 +1533,35 @@ def _updateUponMaterialMetadataChange(self) -> None:
         with postponeSignals(*self._getContainerChangedSignals(), compress = CompressTechnique.CompressPerParameterValue):
             self.updateMaterialWithVariant(None)
             self._updateQualityWithMaterial()
+
+    ##  Get default firmware file name if one is specified in the firmware
+    @pyqtSlot(result = str)
+    def getDefaultFirmwareName(self):
```

Comment:
I guess this function can be moved to `GlobalStack`? Prevent `MachineManager` from uncontrollable growth.

Reply:
Good point, fixed in bc52830